test/unit/com/unifina/datasource/DataSourceEventQueueSpec.groovy

```diff
@@ -147,4 +146,43 @@ class DataSourceEventQueueSpec extends Specification {
 		})
 	}
 
+	void "timer drift"() {
+		DataSourceEventQueue queue = createQueue()
+		int eventsProcessed = 0
+
+		when:
+		Thread consumerThread = Thread.start {
+			queue.start()
+		}
+
+		Thread producerThread = Thread.start {
+			queue.enqueue(new Event<Integer>(1, new Date(10000), new Consumer<Integer>() {
+				@Override
+				void accept(Integer integer) {
+					eventsProcessed++
+				}
+			}))
+			queue.enqueue(new Event<Integer>(2, new Date(20000), new Consumer<Integer>() {
+				@Override
+				void accept(Integer integer) {
+					eventsProcessed++
+				}
+			}))
+		}
+
```

Comment:
Would be good to test that each second in between these two events is being reported. So you could additionally do something like (untested code):

```
ITimeListener timeListener = Mock(ITimeListener)
timeListener.tickRateInSec() >> 1
queue.addTimeListener(timeListener)
```

And then assert that the `timeListener.setTime(...)` gets called N times (I guess 11 times in this case)

The timestamps it should be called with are:

```
10000
11000
...
20000
```